### PR TITLE
COS-246 Change retrieving of collection operator(get from bucket instead of cluster)

### PR DESCRIPTION
**Pull request by tigrankhojoyan**
…tead of cluster)

### CODE REVIEW

**Review by DaimonCool — APPROVED** [review]

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/statement/CreateQueryIndexStatement.java [hunk 1]
```diff
@@ -34,11 +33,11 @@ public class CreateQueryIndexStatement extends CouchbaseStatement {
...
     @Override
     public void execute(ClusterOperator clusterOperator) {
-        Collection collection = clusterOperator.getBucketOperator(keyspace.getBucket())
-                .getCollection(keyspace.getCollection(), keyspace.getScope());
         CreateQueryIndexOptions options = CreateQueryIndexOptions.createQueryIndexOptions()
                 .deferred(deferred)
                 .numReplicas(numReplicas);
-        clusterOperator.getCollectionOperator(collection).createQueryIndex(indexName, fields, options);
+        clusterOperator.getBucketOperator(keyspace.getBucket())
+                .getCollectionOperator(keyspace.getCollection(), keyspace.getScope())
```
Comment (konstantin-umanets) [inline liquibase-couchbase/src/main/java/liquibase/ext/couchbase/statement/CreateQueryIndexStatement.java:40]: Could we add a version of this method that consumes keyspace to get collection and scope from it?

File: liquibase-couchbase/src/test/java/common/operators/TestClusterOperatorTest.java [hunk 2]
```diff
@@ -95,7 +96,9 @@ public BucketSettings answer(InvocationOnMock invocation) {
 
     @Test
     void should_create_collection_primary_index() {
-        testClusterOperator.getCollectionOperator(collection).createCollectionPrimaryIndex(null);
+        testClusterOperator.getBucketOperator(TEST_BUCKET)
+                .getCollectionOperator(TEST_COLLECTION, TEST_SCOPE)
+                .createCollectionPrimaryIndex(null);
```
Comment (konstantin-umanets) [inline liquibase-couchbase/src/test/java/common/operators/TestClusterOperatorTest.java:101]: In all the cases in this PR we use this method with null value as a parameter. Is it possible to add the method version without parameters (not to provide a useless null every time)?

File: liquibase-couchbase/src/main/java/liquibase/ext/couchbase/statement/CreateQueryIndexStatement.java (hunk 1 above)
Comment (tigrankhojoyan) [reply liquibase-couchbase/src/main/java/liquibase/ext/couchbase/statement/CreateQueryIndexStatement.java:40]: Yes, we can, but in bucket operator we don't need name of bucket, and I think getting explicitly via collection name and scope name is more readable. What you think?

File: liquibase-couchbase/src/test/java/liquibase/ext/couchbase/operator/ClusterOperatorTest.java [hunk 3]
```diff
@@ -216,7 +218,10 @@ void should_return_true_if_collection_index_exist() {
         when(collectionQueryIndexManager.getAllIndexes()).thenReturn(Collections.singletonList(queryIndex));
         when(queryIndex.name()).thenReturn(TEST_INDEX);
 
-        boolean result = clusterOperator.getCollectionOperator(collection).collectionIndexExists(TEST_INDEX);
+        boolean result = clusterOperator
```
Comment (konstantin-umanets) [inline liquibase-couchbase/src/test/java/liquibase/ext/couchbase/operator/ClusterOperatorTest.java:221]: Looks like we could extract `clusterOperator
                .getBucketOperator(TEST_BUCKET)
                .getCollectionOperator(TEST_COLLECTION, TEST_SCOPE)` to class constant, could we?

File: liquibase-couchbase/src/test/java/integration/statement/DropPrimaryIndexStatementIT.java [hunk 4]
```diff
@@ -50,9 +45,9 @@ void Should_drop_Primary_index_by_name() {
     void Should_drop_primary_index_for_specific_keyspace() {
         cluster.waitUntilReady(CLUSTER_READY_TIMEOUT);
         keyspace = keyspace(bucketName, scopeName, collectionName);
-        Collection collection = clusterOperator.getBucketOperator(bucketName)
-                .getCollection(collectionName, scopeName);
-        clusterOperator.getCollectionOperator(collection).createCollectionPrimaryIndex(null);
+        clusterOperator.getBucketOperator(bucketName)
+                .getCollectionOperator(collectionName, scopeName)
+                .createCollectionPrimaryIndex(null);
```
Comment (dmitry-shanko) [inline liquibase-couchbase/src/test/java/integration/statement/DropPrimaryIndexStatementIT.java:50]: I think it should be `createPrimaryIndex()` method call as it plays the same role as `.createCollectionPrimaryIndex(null)`

**Review by romchellis — APPROVED** [review]

**Review by CountRick — APPROVED** [review]

**Pull request merged**